### PR TITLE
Merge jakarta.web.beans and web.beans

### PR DESCRIPTION
**Pull request by matthiasblaesing**
The jakarta.web.beans module was created as a copy of the web.beans
module, just using the new namespaces after the death of JavaEE and the
inception of JakartaEE. Other modules followed a different route, where
the existing implementations were extended to support both the old and
new XML/package namespaces.

This becomes a problem, when it hits the UI as the user then does not
know what to do (which template/menu entry to choose and "what the heck
is JakartaEE and why is that not JavaEE anymore?!"). What is more
codepaths are duplicated and fixes for one area are highly likely to be
relevant for the other also.

This change integrates the unittests from the jakarta.web.beans project,
if they actually check jakarta* functions and modifies the codepaths, so
that both namespaces are supported.

This PR also fixes and reenables the CI/CD integration of the web.beans tests.

Closes: #7918
Closes: #6875

### CODE REVIEW

File: enterprise/web.beans/src/org/netbeans/modules/web/beans/analysis/analyzer/AnnotationUtil.java
Comment (mbien) [inline enterprise/web.beans/src/org/netbeans/modules/web/beans/analysis/analyzer/AnnotationUtil.java:1]: this class was retrofitted to use varargs at some point, but not all methods have them. I believe the changeset can be greately reduced by adding this:

    
```java
    public static boolean hasAnnotation(Element element, CompilationInfo info, String... annotations) {
        return getAnnotationMirror(element, info, annotations) != null;
    }
```
this would get rid of many if-blocks.

consider also _removing_ those two:
```java
    public static AnnotationMirror getAnnotationMirror(Element element,
            String annotation,CompilationInfo info )
    {
        return getAnnotationMirror(element, info, annotation);
    }
```
```java
    public static boolean hasAnnotation(Element element, String annotation,
            CompilationInfo info )
    {
        return getAnnotationMirror(element, info)!=null;
    }
```
since it already exists as varargs variant with swapped params. This isn't public API I believe - should be fine.

**Review by mbien — COMMENTED** [review]
i think this makes sense since the copied modules would likely never diverge enough to warrant having two impls. Only looked through this on github so far.

**Comment by matthiasblaesing** [comment]
@mbien thanks for the suggestion. Pushed an update, might squashed with b4c6e558abf8bedd57c2174b424b1c075069a7d0 before merge.

**Review by mbien — APPROVED** [review]
Looked through everything with a coffee and I think it should be fine.

the comments aren't that important. `List.of()` is immutable and has the advantage that it makes followup `List.copyOf()`'s a no-op. It also doesn't allow null, so it would fail early.

**Comment by matthiasblaesing** [comment]
@mbien thank you. I updated the PR and plan to squash 24e6f9b into [b4c6e55](https://github.com/apache/netbeans/pull/7958/commits/b4c6e558abf8bedd57c2174b424b1c075069a7d0) before merge.

**Comment by matthiasblaesing** [comment]
Squashed and rebased onto master, resolving the merge conflict. I removed the reference to `jakarta.web.beans` from the lsp build and requested a full test run with this.

I intent to merge this shortly when/if tests are green.

**Comment by mbien** [comment]
the failing test seems to be a real issue, but unrelated to this PR. Have to check when this started.

**Comment by matthiasblaesing** [comment]
@mbien thanks for having a look. No need to apologize, flaky tests are ugly. I ran the tests locally and apart from locale problems (since a few releases javac localizes messages) and the excessive runtime, everything seems to be ok. So I'll merge.

**Pull request merged**
